### PR TITLE
docs(s3): better explain behavior of BUCKET_OWNER_PREFERRED

### LINKED ISSUE #28866: (ObjectOwnership): (Update documentation to be more explicit to match S3 documentation)
### Describe the issue

When we created a S3 bucket via Bucket [construct](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_s3.Bucket.html) we used the ```BUCKET_OWNER_PREFERRED``` setting for [objectOwnership](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_s3.ObjectOwnership.html#bucket_owner_preferred) but we found Objects uploaded to the bucket DIDN’T change ownership to the bucket owner and we had problems downloading the objects.

Then we found out if the objects are being uploaded to the bucket without ```bucket-owner-full-control``` canned ACL, the bucket owner won’t have permissions see S3 [documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/about-object-ownership.html) but the CDK documentation doesn’t mention it explicitly, It currently has

```
BUCKET_OWNER_PREFERRED

Objects uploaded to the bucket change ownership to the bucket owner .

```

We had to explicitly change the enum to ```BUCKET_OWNER_ENFORCED``` or upload the objects with  the canned acl see support [article](https://repost.aws/knowledge-center/s3-bucket-owner-full-control-acl)

So we propose the documentation be updated to be more explicit similar to S3 documentation.

```
The bucket owner owns and has full control over new objects that other accounts write to the bucket with the bucket-owner-full-control canned ACL. 
```
 We used CDK(Typescript) but other languages could be updated as well.
 
 Discussed with Kaizen before entering this PR.

Thank you

### Links

https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_s3.ObjectOwnership.html#bucket_owner_preferred

https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_s3/ObjectOwnership.html

https://docs.aws.amazon.com/cdk/api/v2/java/software/amazon/awscdk/services/s3/ObjectOwnership.html

https://pkg.go.dev/github.com/aws/aws-cdk-go/awscdk/v2/awss3#ObjectOwnership

https://pkg.go.dev/github.com/aws/aws-cdk-go/awscdk/v2/awss3#ObjectOwnership

https://docs.aws.amazon.com/cdk/api/v2/dotnet/api/Amazon.CDK.AWS.S3.ObjectOwnership.html

### PR DESCRIPTION
**Pull request by jramacha**
Update the enum ObjectOwnership docs to be more explicit. As mentioned [here](https://github.com/aws/aws-cdk/issues/28866) the documentation is not explicit but in S3 [documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/about-object-ownership.html) its explicit. We had discussed with Kaizen in [issue](https://github.com/aws/aws-cdk/issues/28866) before entering this PR/Issue.

Closes #28866 .

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-s3/lib/bucket.ts [hunk 1]
```diff
@@ -1291,7 +1291,7 @@ export enum ObjectOwnership {
    */
   BUCKET_OWNER_PREFERRED = 'BucketOwnerPreferred',
   /**
-   * The uploading account will own the object.
+   * The bucket owner will own the object if the object is uploaded with the bucket-owner-full-control canned ACL. Without this setting and canned ACL, the object is uploaded and remains owned by the uploading account.
```
Comment (kaizencc) [inline packages/aws-cdk-lib/aws-s3/lib/bucket.ts:1294]: The issue mentions `BUCKET_OWNER_PREFERRED`, but we are updating `OBJECT_WRITER`'s docstring here. Did you mean to update `BUCKET_OWNER_PREFERRED` instead?

Comment (jramacha) [reply packages/aws-cdk-lib/aws-s3/lib/bucket.ts:1294]: Oops.. Thanks for catching it ,I'll update it.

**Review by kaizencc — APPROVED** [review]
Thanks!

File: packages/aws-cdk-lib/aws-s3/lib/bucket.ts [hunk 2]
```diff
@@ -1287,7 +1287,9 @@ export enum ObjectOwnership {
    */
   BUCKET_OWNER_ENFORCED = 'BucketOwnerEnforced',
   /**
-   * Objects uploaded to the bucket change ownership to the bucket owner .
+   * The bucket owner will own the object if the object is uploaded with 
+   * the bucket-owner-full-control canned ACL. Without this setting and 
+   * canned ACL, the object is uploaded and remains owned by the uploading account.
```
Comment (kaizencc) [inline packages/aws-cdk-lib/aws-s3/lib/bucket.ts:1292]:
```suggestion
   * The bucket owner will own the object if the object is uploaded with
   * the bucket-owner-full-control canned ACL. Without this setting and
   * canned ACL, the object is uploaded and remains owned by the uploading account.
```

**Review by kaizencc — APPROVED** [review]

**Pull request merged**